tests/test_config_space.py

```diff
@@ -0,0 +1,175 @@
+lpmethod {0, 1, 2, 3, 4, 5, 6} [0]
+mip_cuts_cliques {-1, 0, 1, 2, 3} [0]
+mip_cuts_covers {-1, 0, 1, 2, 3} [0]
+mip_cuts_disjunctive {-1, 0, 1, 2, 3} [0]
+mip_cuts_flowcovers {-1, 0, 1, 2} [0]
+mip_cuts_gomory {-1, 0, 1, 2} [0]
+mip_cuts_gubcovers {-1, 0, 1, 2} [0]
+mip_cuts_implied {-1, 0, 1, 2} [0]
+mip_cuts_mcfcut {-1, 0, 1, 2} [0]
+mip_cuts_mircut {-1, 0, 1, 2} [0]
+mip_cuts_pathcut {-1, 0, 1, 2} [0]
+mip_cuts_zerohalfcut {-1, 0, 1, 2} [0]
+mip_limits_aggforcut [0, 10] [3]i
+mip_limits_cutpasses {-1, 0, 1, 4, 16, 64} [0]
+mip_limits_cutsfactor [1.0, 16.0] [4.0]l
+mip_limits_gomorycand [50, 800] [200]il
+mip_limits_gomorypass {0, 1, 4, 16, 64} [0]
+mip_limits_submipnodelim [125, 2000] [500]il
+mip_ordertype {0, 1, 2, 3} [0]
+mip_strategy_backtrack {0.9, 0.99, 0.999, 0.9999, 0.99999, 0.999999} [0.9999]
+mip_strategy_bbinterval [1, 1000] [7]il
+mip_strategy_branch {-1, 0, 1} [0]
+mip_strategy_dive {0, 1, 2, 3} [0]
+mip_strategy_file {0, 1} [1]
+mip_strategy_fpheur {-1, 0, 1, 2} [0]
+mip_strategy_heuristicfreq {-1, 0, 5, 10, 20, 40, 80} [0]
+mip_strategy_lbheur {yes, no} [no]
+mip_strategy_nodeselect {0, 1, 2, 3} [1]
+mip_strategy_presolvenode {-1, 0, 1, 2} [0]
+mip_strategy_probe {-1, 0, 1, 2, 3} [0]
+mip_strategy_rinsheur {-1, 0, 5, 10, 20, 40, 80} [0]
+mip_strategy_search {0, 1, 2} [0]
+mip_strategy_startalgorithm {0, 1, 2, 3, 4, 5, 6} [0]
+mip_strategy_subalgorithm {0, 1, 2, 3, 4, 5} [0]
+mip_strategy_variableselect {-1, 0, 1, 2, 3, 4} [0]
+network_netfind {1, 2, 3} [2]
+network_pricing {0, 1, 2} [0]
+preprocessing_aggregator {-1, 0, 1, 4, 16, 64} [-1]
+preprocessing_boundstrength {-1, 0, 1} [-1]
+preprocessing_coeffreduce {0, 1, 2} [2]
+preprocessing_dependency {-1, 0, 1, 2, 3} [-1]
+preprocessing_dual {-1, 0, 1} [0]
+preprocessing_fill [2, 40] [10]il
+preprocessing_linear {0, 1} [1]
+preprocessing_numpass {-1, 0, 1, 4, 16, 64} [-1]
+preprocessing_reduce {0, 1, 2, 3} [3]
+preprocessing_relax {-1, 0, 1} [-1]
+preprocessing_repeatpresolve {-1, 0, 1, 2, 3} [-1]
+preprocessing_symmetry {-1, 0, 1, 2, 3, 4, 5} [-1]
+read_scale {-1, 0, 1} [0]
+sifting_algorithm {0, 1, 2, 3, 4} [0]
+simplex_crash {-1, 0, 1} [1]
+simplex_dgradient {0, 1, 2, 3, 4, 5} [0]
+simplex_limits_perturbation {0, 1, 4, 16, 64} [0]
+simplex_limits_singularity [2, 40] [10]il
+simplex_perturbation_switch {no, yes} [no]
+simplex_pgradient {-1, 0, 1, 2, 3, 4} [0]
+simplex_pricing {0, 1, 4, 16, 64} [0]
+simplex_refactor {0, 4, 16, 64, 256} [0]
+simplex_tolerances_markowitz [0.0001, 0.5] [0.01]l
+mip_limits_strongcand [2, 40] [10]il
+mip_limits_strongit {0, 1, 4, 16, 64} [0]
+mip_strategy_order {yes, no} [yes]
+perturbation_constant [1e-08, 0.0001] [1e-06]l
+
+mip_strategy_order | mip_ordertype in {1, 2, 3}
+mip_limits_strongcand | mip_strategy_variableselect in {3}
+mip_limits_strongit | mip_strategy_variableselect in {3}
+perturbation_constant | simplex_perturbation_switch in {yes}
+'''
+
+cs = pcs.read(io.StringIO(pcss))
+
+params = convert_from_config_space(cs)
```

Comment:
This seems much more complicated than calling https://mlopez-ibanez.github.io/irace/reference/read_pcs_file.html  Why is necessary to do this?

Reply:
The ConfigSpace seems to have made a new format called pcs_new https://automl.github.io/ConfigSpace/main/api/serialization.html#module-ConfigSpace.read_and_write.pcs_new. This would allow it to read that too because `ConfigSpace` will handle the parsing. Also when parsing the conditions, ConfigSpace tries to do some complicated thing to join the conditions with and statement and check if they logically make sense.

Reply:
Is this new format the one used by the experiments with the surrogates? Why do we need the new format?